packages/bibtex/init.lua

```diff
@@ -48,14 +106,88 @@ local parseBibtex = function (fn)
    if not t or not t[1] or t.id ~= "document" then
       SU.error("Error parsing bibtex")
    end
-   local entries = {}
    for i = 1, #t do
       if t[i].id == "entry" then
          local ent = t[i][1]
-         entries[ent.label] = { type = ent.type, attributes = ent[1] }
+         local entry = { type = ent.type, attributes = ent[1] }
+         if biblio[ent.label] then
+            SU.warn("Duplicate entry key '" .. ent.label .. "', picking the last one")
+         end
+         biblio[ent.label] = consolidateEntry(entry, ent.label)
+      end
+   end
+end
+
+--- Copy fields from the parent entry to the child entry.
+-- BibLaTeX/Biber have a complex inheritance system for fields.
+-- This implementation is more naive, but should be sufficient for reasonable
+-- use cases.
+-- @tparam table parent Parent entry
```

Comment:
It warms my cold clammy heart to see function docs and LSP / doc generation hints. In general. In this case I do kind of wonder what the point is on a local function that isn't exported for use anywhere outside of this file. Or did I miss something?

Reply:
I've the feeling that the bibtex package, providing `\cite`, `\reference`, `\loadbibliography` etc. is somewhat ill-named, it should be bibliography, and support for bibliography formats should be provided in their own files. Besides `.bib`, one could want to provide references in RIS, Refer, EndNotes... or even CSL-JSON, eventually...). It's a bit soon to say, maybe, and I didn't want to split things too early here.

Whatever, having the LSP instructed for type safety and parameter meanings (regardless of an eventual doc generation) is always nice, no?

Reply:
Yes I agree this should be a bibliography package with somewhat modular support for loaders.

As for the docstring hints, sure, and it should affect editors editing *that file*. Just be aware it won't get published since it isn't a public function.

Reply:
The TODO item in this discussion is now #2076.